### PR TITLE
Updates recording dialog.

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

**Comment by damencho** [comment]
Web screenshots:
screenshot 2019-03-05 at 17 45 18
screenshot 2019-03-05 at 17 44 43
screenshot 2019-03-05 at 17 44 34
screenshot 2019-03-05 at 17 44 06
screenshot 2019-03-05 at 17 43 50

**Comment by damencho** [comment]
Mobile screenshots:
screenshot_20190305-180215
screenshot_20190305-174823
screenshot_20190305-174832
screenshot_20190305-180344
screenshot_20190305-180339

File: react/features/recording/components/Recording/StartRecordingDialogContent.js [hunk 1]
```diff
@@ -107,10 +155,47 @@ class StartRecordingDialogContent extends Component<Props> {
      * @returns {React$Component}
      */
     _renderNoIntegrationsContent() {
+
+        if (!this.props.fileRecordingsServiceEnabled
+            && this.props.integrationsEnabled) {
+            return null;
```
Comment (virtuacoplenny) [inline react/features/recording/components/Recording/StartRecordingDialogContent.js]: I'm a little confused about what this method is doing versus. This method specifically is for jitsi's file recording service? But it also handles displaying integrations?

File: react/features/recording/components/Recording/StartRecordingDialogContent.js [hunk 2]
```diff
@@ -164,18 +284,49 @@ class StartRecordingDialogContent extends Component<Props> {
...
+     */
+    _onDropboxSwitchChange() {
+        // act like group, cannot toggle off
+        if (this.state.dropboxRecordingSelected) {
+            return;
+        }
+
+        this.setState({
+            jitsiRecordingSelected: false,
+            dropboxRecordingSelected: true
```
Comment (virtuacoplenny) [inline react/features/recording/components/Recording/StartRecordingDialogContent.js]: What about instead setting an enum for the selected service? Would that be tricky or even better at all?

File: react/features/base/react/components/native/Button.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <TouchableOpacity
+                onPress = { this.props.onValueChange } >
+                <Text style = { this.props.style }>
+                    { this.props.children }
+                </Text>
```
[the remark this answers is not in the document]
Comment (damencho) [reply react/features/base/react/components/native/Button.js]: I see same example MyAppText in https://facebook.github.io/react-native/docs/text:
<Text style={{fontSize: 20}}>{this.props.children}</Text>

File: react/features/base/react/components/web/Button.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+     * @returns {ReactElement}
+     */
+    render() {
+        const { onValueChange } = this.props;
+
+        return (
+            <Button
+                appearance = 'primary'
+                onClick = { onValueChange }
+                type = 'button'>
```
[the remark this answers is not in the document]
Comment (damencho) [reply react/features/base/react/components/web/Button.js]: Nope :) if it happens to remove @atlaskit those will change probably ... and then when making that change we may use this abstraction for native and web and encapsulate the dependency like atlaskit ...

File: react/features/recording/components/Recording/StartRecordingDialogContent.js (hunk 1 above)
Comment (damencho) [reply react/features/recording/components/Recording/StartRecordingDialogContent.js]: Well, these are the conditions when to show/hide the non-integration part.
Show it when
- fileRecordingsServiceEnabled is enabled
- or show it when there are no integrations
I will reverse it then and will add a comment.

File: config.js [hunk 5]
```diff
@@ -175,6 +175,9 @@ var config = {
     // dropbox: {
     //     appKey: '<APP_KEY>' // Specify your app key here.
     // },
+    // When integrations like dropbox are enabled only that will be shown,
+    // by enabling fileRecordingsServiceEnabled, we force to show both
```
[the remark this answers is not in the document]
Comment (damencho) [reply config.js]: Changed the doc.

File: react/features/recording/components/Recording/StartRecordingDialogContent.js (hunk 1 above)
Comment (damencho) [reply react/features/recording/components/Recording/StartRecordingDialogContent.js]: Done.

File: react/features/recording/components/Recording/StartRecordingDialogContent.js (hunk 2 above)
Comment (damencho) [reply react/features/recording/components/Recording/StartRecordingDialogContent.js]: I've changed it to using enums.

File: react/features/recording/components/Recording/StartRecordingDialogContent.js [hunk 6]
```diff
@@ -164,18 +278,49 @@ class StartRecordingDialogContent extends Component<Props> {
...
+     * Handler for onValueChange events from the Switch component.
+     *
+     * @returns {void}
+     */
+    _onDropboxSwitchChange() {
+        // act like group, cannot toggle off
+        if (this.state.selectedRecordingService
+                === RECORDING_TYPES.DROPBOX) {
+            return;
+        }
```
Comment (virtuacoplenny) [inline react/features/recording/components/Recording/StartRecordingDialogContent.js]: Might be easier to let the setState happen again.

**Pull request merged**